tests/e2e/system_test/test_cr_resources_validation.py

```diff
@@ -22,47 +22,93 @@ class TestCRRsourcesValidation(E2ETest):
         self.object_name_to_delete = ""
-        network_fence_yaml = os.path.join(
-            constants.TEMPLATE_CSI_ADDONS_DIR, "NetworkFence.yaml"
-        )
-        res = run_oc_command(cmd=f"create -f {network_fence_yaml}")
+
+    def cr_resource_not_editable(
+        self, cr_object_kind, yaml_name, patches_non_editable, patches_editable
+    ):
+        """
+        Test that cr object is not editable once created
+
+        Args:
+            cr_object_kind (str): cr object kind
+            yaml_name (str): name of the yaml file from which the object is to be created
+            patches_non_editable (dict, of str: str): patches to be applied by 'oc patch' command. These patches should
+                    have no effect. If such a patch is sucessfully applied, the test should fail
+            patches_editable (dict, of str: str): patches to be applied by 'oc patch' command. These patches should
+                    have an effect. If such a patch is not sucessfully applied, the test should fail
+
+        """
+        cr_resource_yaml = os.path.join(constants.TEMPLATE_CSI_ADDONS_DIR, yaml_name)
+        res = run_oc_command(cmd=f"create -f {cr_resource_yaml}")
         assert (
             ERRMSG not in res[0]
-        ), f"Failed to create resource Network Fence from yaml file {network_fence_yaml}, got result {res}"
+        ), f"Failed to create resource {cr_object_kind} from yaml file {cr_resource_yaml}, got result {res}"
 
-        network_fence_name = res[0].split()[0]
-        self.object_name_to_delete = network_fence_name
+        cr_resource_name = res[0].split()[0]
+        self.object_name_to_delete = cr_resource_name
 
-        network_fence_original_yaml = run_oc_command(
-            f"get {network_fence_name} -o yaml"
-        )
+        cr_resource_original_yaml = run_oc_command(f"get {cr_resource_name} -o yaml")
 
-        patches = {  # dictionary: patch_name --> patch
-            "apiVersion": {"apiVersion": "csiaddons.openshift.io/v1alpha2"},
-            "kind": {"kind": "newNetworkFence"},
-            "generation": '{"metadata": {"generation": 6456456 }}',
-            "creationTime": '{"metadata": {"creationTimestamp": "2022-04-24T19:39:54Z" }}',
-            "name": '{"metadata": {"name": "newName" }}',
-            "uid": '{"metadata": {"uid": "897b3c9c-c1ce-40e3-95e6-7f3dadeb3e83" }}',
-            "secret": '{"spec": {"secret": {"name": "new_secret"}}}',
-            "cidrs": '{"spec": {"cidrs": {"10.90.89.66/32", "11.67.12.42/32"}}}',
-            "fenceState": '{"spec": {"fenceState": "NotFenced"}}',
-            "driver": '{"spec": {"driver": "example.new_driver"}}',
-            "new_property": '{"spec": {"new_property": "new_value"}}',
-        }
+        # test that all non editable properties are really not editable
+        non_editable_properties_errors = {}
+        cr_resource_prev_yaml = cr_resource_original_yaml
+        for patch in patches_non_editable:
```

Comment:
wouldn't it be possible to create file templates and use them instead of writing files on the go?

Reply:
I don't think we are writing files on the go. The patch is applied to the created cr object ( this is similar to "oc edit ______" . 
Also I've tried number of different methods to update the object and this seems the only one working. Therefore I'd leave it as is. Thank you.